models/question.ts

```diff
@@ -0,0 +1,41 @@
+import * as uuid from 'uuid';
+
+export enum QuestionType {
+  TEXT = 'text',
+}
+
+export type QuestionData = TextQuestionData;
+
+interface QuestionBase {
+  id: string;
+  type: QuestionType;
+  data: QuestionData;
+  createdAt: string;
+}
+
+export interface BaseQuestionData {
+  question: string;
+  required: boolean;
+  type: QuestionType;
+}
+
+export interface TextQuestionData extends BaseQuestionData {
+  type: QuestionType.TEXT;
+}
+
+export interface FormQuestion {
+  id: string;
+  type: QuestionType;
+  data: QuestionData;
+  createdAt: string;
+}
+
+export type QuestionCreateParams = TextQuestionData;
+
+export class TextQuestion implements QuestionBase {
```

Comment:
How do you like the idea of making `QuestionBase` a class that initializes `createdAt` and `id` then inheriting it so for each question type we won't have to specify those, it's only two lines, so I'm good with either, I suppose that could grow though.

Reply:
I think this is too soon an optimization for now - lets keep an eye on it and see if it's worth it later.

Reply:
eh i took another look, it still needs the interface bc type and data are never supposed to be optional, but checkout what i have now